src/CMakeLists.txt

```diff
@@ -1,6 +1,43 @@
-# Copyright (c) 2023 The Bitcoin Core developers
+# Copyright (c) 2023-present The Bitcoin Core developers
 # Distributed under the MIT software license, see the accompanying
-# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+# file COPYING or https://opensource.org/license/mit/.
+
+#=============================
+# bitcoin_crypto library
+#=============================
+# This subdirectory includes compiler flag checks, which print
+# their results. Therefore, we place it before the secp256k1
+# subtree. This ensures that both summaries—secp256k1's and
+# Bitcoin Core's—are displayed one after the other.
+add_subdirectory(crypto)
```

Comment:
Looks like this shuffling broke most builds. Rather than moving this around, why can't we just do the secp256k1 configuration last?

Reply:
The `secp256k1` target, as a CMake's abstraction, must be known before its usage in statements like https://github.com/hebasto/bitcoin/blob/74bf342e05d409c800edae2f69fcea6a9c38a4e9/src/CMakeLists.txt#L94-L99

Reply:
By last, I mean:
```diff
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 18d07e56ef..d2ef9860d6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -8,14 +8,35 @@ include(AddWindowsResources)
 configure_file(${CMAKE_SOURCE_DIR}/cmake/bitcoin-config.h.in config/bitcoin-config.h @ONLY)
 include_directories(${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
 
-#=============================
-# bitcoin_crypto library
-#=============================
-# This subdirectory includes compiler flag checks, which print
-# their results. Therefore, we place it before the secp256k1
-# subtree. This ensures that both summaries—secp256k1's and
-# Bitcoin Core's—are displayed one after the other.
+# After the transition from Autotools to CMake, the obj/ subdirectory
+# could be dropped as its only purpose was to separate a generated header
+# from source files.
+add_custom_target(generate_build_info
+  BYPRODUCTS ${PROJECT_BINARY_DIR}/src/obj/build.h
+  COMMAND ${CMAKE_COMMAND} -E make_directory ${PROJECT_BINARY_DIR}/src/obj
+  COMMAND ${CMAKE_COMMAND} -DBUILD_INFO_HEADER_PATH=${PROJECT_BINARY_DIR}/src/obj/build.h -DSOURCE_DIR=${PROJECT_SOURCE_DIR} -P ${CMAKE_SOURCE_DIR}/cmake/script/GenerateBuildInfo.cmake
+  COMMENT "Generating obj/build.h"
+  VERBATIM
+)
+add_library(bitcoin_clientversion OBJECT EXCLUDE_FROM_ALL
+  clientversion.cpp
+)
+target_compile_definitions(bitcoin_clientversion
+  PRIVATE
+    HAVE_BUILD_INFO
+)
+target_link_libraries(bitcoin_clientversion
+  PRIVATE
+    core_interface
+)
+add_dependencies(bitcoin_clientversion generate_build_info)
+
 add_subdirectory(crypto)
+add_subdirectory(univalue)
+add_subdirectory(util)
+if(WITH_MULTIPROCESS)
+  add_subdirectory(ipc)
+endif()
 
 #=============================
 # secp256k1 subtree
@@ -46,35 +67,6 @@ add_subdirectory(secp256k1 EXCLUDE_FROM_ALL)
 set_target_properties(secp256k1 PROPERTIES EXPORT_COMPILE_COMMANDS OFF)
 string(APPEND CMAKE_C_COMPILE_OBJECT " ${APPEND_CPPFLAGS} ${APPEND_CFLAGS}")
 
-# After the transition from Autotools to CMake, the obj/ subdirectory
-# could be dropped as its only purpose was to separate a generated header
-# from source files.
-add_custom_target(generate_build_info
-  BYPRODUCTS ${PROJECT_BINARY_DIR}/src/obj/build.h
-  COMMAND ${CMAKE_COMMAND} -E make_directory ${PROJECT_BINARY_DIR}/src/obj
-  COMMAND ${CMAKE_COMMAND} -DBUILD_INFO_HEADER_PATH=${PROJECT_BINARY_DIR}/src/obj/build.h -DSOURCE_DIR=${PROJECT_SOURCE_DIR} -P ${CMAKE_SOURCE_DIR}/cmake/script/GenerateBuildInfo.cmake
-  COMMENT "Generating obj/build.h"
-  VERBATIM
-)
-add_library(bitcoin_clientversion OBJECT EXCLUDE_FROM_ALL
-  clientversion.cpp
-)
-target_compile_definitions(bitcoin_clientversion
-  PRIVATE
-    HAVE_BUILD_INFO
-)
-target_link_libraries(bitcoin_clientversion
-  PRIVATE
-    core_interface
-)
-add_dependencies(bitcoin_clientversion generate_build_info)
-
-add_subdirectory(univalue)
-add_subdirectory(util)
-if(WITH_MULTIPROCESS)
-  add_subdirectory(ipc)
-endif()
-
 
 # Stable, backwards-compatible consensus functionality.
 add_library(bitcoin_consensus STATIC EXCLUDE_FROM_ALL
```

Reply:
Accepted.

Reply:
Looks like this was broken again? Qt compiler checks and output appears after secp256k1 configure.